blog/2024-06-02-kqir-query-engine/index.md

```diff
@@ -0,0 +1,222 @@
+---
+slug: kqir-query-engine
+title: "KQIR: a query engine for Apache Kvrocks that supports both SQL and RediSearch queries"
+authors: [twice]
+---
+
+## Intro
+
+TL;DR:
+
+![ftsearch-demo](./images/ftsearch-demo.png)
+
+Pretty cool, right? Let's dive in!
+
+<!--truncate-->
+
+### Apache Kvrocks
+
+[Apache Kvrocks](https://kvrocks.apache.org/) is a [Redis](https://redis.io/)-compatible database built on [RocksDB](https://rocksdb.org/).
+
+It supports [the RESP protocol](https://redis.io/docs/latest/develop/reference/protocol-spec/) (version 2 and 3) and [a wide range of Redis commands](/docs/supported-commands), encompassing core data structures like Strings, Sets, Hashes, Sorted Sets, Stream, GEO, as well as Lua Scripts, Transactions, [Functions](https://redis.io/docs/latest/develop/interact/programmability/functions-intro/) and even [BloomFilter](https://redis.io/docs/latest/develop/data-types/probabilistic/bloom-filter/), [JSON](https://redis.io/docs/latest/develop/data-types/json/) from the Redis Stack.
+
+Unlike Redis which stores data in memory, Kvrocks persists data on disk for improved storage capabilities without being constrained by machine memory limit.
+
+### The capability to query
+
+In recent decades, NoSQL databases have gained prominence over traditional databases for their superior performance, scalability, and versatility across various industries.
```

Comment:
```suggestion
In recent years, NoSQL databases have become more popular than traditional databases because they perform better, scale easily, and are more flexible for different industries.
```

Ditto. Use simple words.